app/Http/Controllers/CRUD/CityController.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace App\Http\Controllers\CRUD;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\CRUD\StoreCityRequest;
+use App\Models\CRUD\City;
+use Illuminate\Http\Request;
+use Redirect;
+use View;
+
+class CityController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $cities = City::all();
+        return View('crud.city.list')
+        ->withCities($cities);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return View('crud.city.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(StoreCityRequest $request)
+    {
+        City::create($request->all());
+    
+        return Redirect()->route('city.index')
+       ->with('success','City created successfully !');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $condition = array('id' => $id);
+        $city = City::where($condition)->firstOrFail();
+ 
+        return view('crud.city.show')
+        ->withCity($city);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
```

Comment:
For edition, you can replace the prototype to use directly the City Model:
```php
public function edit(City $city)
```
it will query for the model City if found, else it will return  a 404 not found page